packages/core/upload/server/controllers/content-api.js

```diff
@@ -94,9 +94,13 @@ module.exports = {
     const apiUploadFolderService = getService('api-upload-folder');
 
     const apiUploadFolder = await apiUploadFolderService.getAPIUploadFolder();
-    data.fileInfo = data.fileInfo || {};
-    data.fileInfo = Array.isArray(data.fileInfo) ? data.fileInfo : [data.fileInfo];
-    data.fileInfo.forEach(fileInfo => (fileInfo.folder = apiUploadFolder.id));
+
+    if (Array.isArray(files)) {
+      data.fileInfo = data.fileInfo || [];
+      data.fileInfo = files.map((_f, i) => ({ ...data.fileInfo[i], folder: apiUploadFolder.id }));
```

Comment:
I think the spread wont work if `fileInfo` is empty. If you want to do it in one line I guess it would be 
```js
data.fileInfo = files.map((_f, i) => ({ ...(data.fileInfo[i] || {}), folder: apiUploadFolder.id }));
```

Also, line 92, I think you can replace the line by :
```js
const data = await validateUploadBody(body, Array.isArray(files));
```
As you said, something was wrong with `validateUploadBody`

Reply:
Tested the spread and I think it works!

In this case, `data.fileInfo[i]` could either be an object or `undefined`, and spreading `undefined` should be equivalent to spreading an empty object.

I also applied the suggestion on line 92.

Reply:
Indeed, I didn't know we could spread `undefined`! I'm glad to know now :D